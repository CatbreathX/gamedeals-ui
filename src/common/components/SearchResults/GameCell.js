import PropTypes from 'prop-types';

export const GameCell = ({ title, price }) => (
  <div>
    <div>
      {title}
      :
      {price}
    </div>
  </div>
);

GameCell.propTypes = {
  title: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
};
