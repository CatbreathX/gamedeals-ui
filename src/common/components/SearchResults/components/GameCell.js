import {
  DataItem,
  DataCell,
  DataValue, DataCellContainer,
} from 'common/components/SearchResults/components/layout/styles';
import PropTypes from 'prop-types';

export const GameCell = ({
  dealId,
  title,
  normalPrice,
  storeName,
  salePrice,
  thumbnailImage,
  priceSaving,
  metacriticRating,
  steamRatingText,
  dealRating,
}) => (
  <DataCellContainer key={dealId}>
    <DataCell>
      <img src={thumbnailImage} alt={title} />
    </DataCell>

    <DataCell>
      <DataItem className="game_cell_title">{title}</DataItem>
      <DataItem className="game_cell_store">{storeName}</DataItem>
    </DataCell>

    <DataCell>
      <DataItem>Normal Price</DataItem>
      <DataValue>{normalPrice}</DataValue>
    </DataCell>

    <DataCell>
      <DataItem>Sale Price</DataItem>
      <DataValue>{salePrice}</DataValue>
    </DataCell>

    <DataCell>
      <DataItem>Savings</DataItem>
      <DataValue>{priceSaving}</DataValue>
    </DataCell>

    <DataCell>
      <DataItem className="game_cell_metacritic-rating">{metacriticRating}</DataItem>
      <DataItem className="game_cell_steam-rating">{steamRatingText}</DataItem>
      <DataItem className="game_cell_deal-rating">{dealRating}</DataItem>
    </DataCell>

  </DataCellContainer>
);

GameCell.propTypes = {
  dealId: PropTypes.string.isRequired,
  dealRating: PropTypes.string.isRequired,
  metacriticRating: PropTypes.string.isRequired,
  normalPrice: PropTypes.string.isRequired,
  priceSaving: PropTypes.string.isRequired,
  salePrice: PropTypes.string.isRequired,
  steamRatingText: PropTypes.string.isRequired,
  storeName: PropTypes.string.isRequired,
  thumbnailImage: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};
