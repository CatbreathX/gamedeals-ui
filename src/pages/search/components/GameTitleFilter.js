import { Checkbox, FormControlLabel, TextField } from '@mui/material';
import { TitleFilterContainer } from 'pages/search/layout/TitleFilterContainer';
import { Controller, useFormContext } from 'react-hook-form';

export const GameTitleFilter = () => {
  const { control, register } = useFormContext();

  return (

    <TitleFilterContainer>
      <TextField type="text" label="Game Title" {...register('gameTitle')} />
      <FormControlLabel
        control={
          <Controller
            control={control}
            name="exactGameTitle"
            render={({ field }) => {
              const isChecked = field.value === true;
              return <Checkbox {...field} checked={isChecked} />;
            }
            }
          />
        }
        label="Exact Match"
      />
    </TitleFilterContainer>
  );
};
