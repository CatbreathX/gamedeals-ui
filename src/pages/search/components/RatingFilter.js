import { FormControl, MenuItem } from '@mui/material';
import { ReactHookFormSelect } from 'common/components/inputs/ReactHookFormSelect';
import { useFormContext } from 'react-hook-form';

export const RatingFilter = () => {
  const ratings = ['50', '60', '70', '80', '85', '90', '95', '100'];

  const { control } = useFormContext();
  return (
    <FormControl sx={{
      width: 100,
    }}
    >
      <ReactHookFormSelect
        name="gameRatingFilter"
        label="Rating"
        control={control}
        defaultValue=""
      >
        <MenuItem value="">None</MenuItem>
        {ratings.map((rating) => (
          <MenuItem key={`rating${rating}`} value={rating}>
            {rating}
          </MenuItem>
        ))}
      </ReactHookFormSelect>
    </FormControl>
  );
};
