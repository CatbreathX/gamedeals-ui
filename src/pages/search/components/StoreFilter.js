import { Autocomplete, TextField } from '@mui/material';
import { sortBy } from 'lodash/collection';
import { Controller, useFormContext } from 'react-hook-form';
import { useGetStoresQuery } from 'services/gamedealapi';

export const StoreFilter = () => {
  const { control } = useFormContext();

  const {
    data,
    isSuccess,
  } = useGetStoresQuery();

  if (!isSuccess) return null;

  return (
    <>
      {isSuccess &&
        <Controller
          control={control}
          render={({ field }) => (
            <Autocomplete
              sx={{
                width: 350,
              }}
              disablePortal
              id="store-filter"
              options={createOptions(data)}
              isOptionEqualToValue={(option, value) => option.value === value.value}
              renderInput={(params) => (
                <TextField
                  {...params}
                  label="Filter Store"
                />
              )}
              onChange={(e, item) => {
                field.onChange(item?.value);
              }}
            />
          )}
          name="store"
        />
      }
      {!isSuccess &&
        <div />
      }
    </>
  );
};

const createOptions = (stores) => {
  const options = [];
  stores.forEach((store) => {
    if (store.isActive) {
      const option = {
        label: store.storeName,
        value: store.storeID,
      };

      options.push(option);
    }
  });

  const sortedOptions = sortBy(options, ['label']);
  return sortedOptions;
};
