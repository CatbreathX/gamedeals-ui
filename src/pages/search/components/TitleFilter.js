import { Checkbox, FormControlLabel, TextField } from '@mui/material';
import { TitleFilterContainer } from 'pages/search/layout/TitleFilterContainer';
import { Controller, useFormContext } from 'react-hook-form';

export const TitleFilter = () => {
  const { control, register } = useFormContext();

  return (

    <TitleFilterContainer>
      <TextField type="text" label="Title" {...register('title')} />

      <FormControlLabel
        control={
          <Controller
            control={control}
            name="exactTitle"
            render={({ field }) => {
              const isChecked = field.value === true;
              return <Checkbox {...field} checked={isChecked} />;
            }
            }
          />
        }
        label="Exact Match"
      />
    </TitleFilterContainer>
  );
};
