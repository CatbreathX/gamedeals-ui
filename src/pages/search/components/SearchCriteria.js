import { Button } from '@mui/material';
import { Form } from 'common/elements/Form';
import { H1 } from 'common/elements/H1';
import { RatingFilter } from 'pages/search/components/RatingFilter';
import { StoreFilter } from 'pages/search/components/StoreFilter';
import { TitleFilter } from 'pages/search/components/TitleFilter';
import { FilterActions, FilterContainer } from 'pages/search/layout/FilterContainer';
import PropTypes from 'prop-types';
import { FormProvider, useForm } from 'react-hook-form';

export const SearchCriteria = ({ handleFormSubmission }) => {
  const formMethods = useForm();
  const {
    handleSubmit,
  } = formMethods;

  const onFormSubmit = (data) => {
    handleFormSubmission(data);
  };

  return (
    <FormProvider {...formMethods}>
      <Form onSubmit={handleSubmit(onFormSubmit)}>
        <H1>Game Deal Search</H1>

        <FilterContainer>
          <StoreFilter />

          <RatingFilter />
          <TitleFilter />
        </FilterContainer>
        <FilterActions>
          <Button type="submit">Submit</Button>
          <Button type="button">Clear</Button>
        </FilterActions>
      </Form>
    </FormProvider>
  );
};

SearchCriteria.propTypes = {
  handleFormSubmission: PropTypes.func.isRequired,
};
