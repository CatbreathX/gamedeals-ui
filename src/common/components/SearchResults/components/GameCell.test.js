import { GameCell } from 'common/components/SearchResults/components/GameCell';
import React from 'react';
import { renderComponent } from 'unit/componentRenderers';

const defaultProps = {
  dealId: '1',
  dealRating: '4.0',
  metacriticRating: '85',
  normalPrice: '29.99',
  priceSaving: '24.00',
  salePrice: '5.99',
  steamRatingText: 'Mostly Positive',
  storeName: 'Steam',
  thumbnailImage: 'https://BASE_IMAGE/game/1/image',
  title: 'Elite Dangerous',
};

describe('GameCell', () => {
  test('should render correctly', () => {
    const component = renderUI();
    expect(component).toMatchSnapshot();
  });
});

const renderUI = (props = {}) => {
  const mergedProps = { ...defaultProps, ...props };

  const component = renderComponent(
    <GameCell  {...mergedProps} />,
  );

  return component;

};
