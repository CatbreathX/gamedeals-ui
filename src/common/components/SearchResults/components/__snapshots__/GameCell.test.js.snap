// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCell should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-11t2uff-DataCellContainer e16h0enc3"
      >
        <div
          class="css-1vth247-DataCell e16h0enc2"
        >
          <img
            alt="Elite Dangerous"
            src="https://BASE_IMAGE/game/1/image"
          />
        </div>
        <div
          class="css-1vth247-DataCell e16h0enc2"
        >
          <div
            class="game_cell_title css-1e19oy8-DataItem e16h0enc1"
          >
            Elite Dangerous
          </div>
          <div
            class="game_cell_store css-1e19oy8-DataItem e16h0enc1"
          >
            Steam
          </div>
        </div>
        <div
          class="css-1vth247-DataCell e16h0enc2"
        >
          <div
            class="css-1e19oy8-DataItem e16h0enc1"
          >
            Normal Price
          </div>
          <div
            class="css-1mqh484-DataValue e16h0enc0"
          >
            29.99
          </div>
        </div>
        <div
          class="css-1vth247-DataCell e16h0enc2"
        >
          <div
            class="css-1e19oy8-DataItem e16h0enc1"
          >
            Sale Price
          </div>
          <div
            class="css-1mqh484-DataValue e16h0enc0"
          >
            5.99
          </div>
        </div>
        <div
          class="css-1vth247-DataCell e16h0enc2"
        >
          <div
            class="css-1e19oy8-DataItem e16h0enc1"
          >
            Savings
          </div>
          <div
            class="css-1mqh484-DataValue e16h0enc0"
          >
            24.00
          </div>
        </div>
        <div
          class="css-1vth247-DataCell e16h0enc2"
        >
          <div
            class="game_cell_metacritic-rating css-1e19oy8-DataItem e16h0enc1"
          >
            85
          </div>
          <div
            class="game_cell_steam-rating css-1e19oy8-DataItem e16h0enc1"
          >
            Mostly Positive
          </div>
          <div
            class="game_cell_deal-rating css-1e19oy8-DataItem e16h0enc1"
          >
            4.0
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-11t2uff-DataCellContainer e16h0enc3"
    >
      <div
        class="css-1vth247-DataCell e16h0enc2"
      >
        <img
          alt="Elite Dangerous"
          src="https://BASE_IMAGE/game/1/image"
        />
      </div>
      <div
        class="css-1vth247-DataCell e16h0enc2"
      >
        <div
          class="game_cell_title css-1e19oy8-DataItem e16h0enc1"
        >
          Elite Dangerous
        </div>
        <div
          class="game_cell_store css-1e19oy8-DataItem e16h0enc1"
        >
          Steam
        </div>
      </div>
      <div
        class="css-1vth247-DataCell e16h0enc2"
      >
        <div
          class="css-1e19oy8-DataItem e16h0enc1"
        >
          Normal Price
        </div>
        <div
          class="css-1mqh484-DataValue e16h0enc0"
        >
          29.99
        </div>
      </div>
      <div
        class="css-1vth247-DataCell e16h0enc2"
      >
        <div
          class="css-1e19oy8-DataItem e16h0enc1"
        >
          Sale Price
        </div>
        <div
          class="css-1mqh484-DataValue e16h0enc0"
        >
          5.99
        </div>
      </div>
      <div
        class="css-1vth247-DataCell e16h0enc2"
      >
        <div
          class="css-1e19oy8-DataItem e16h0enc1"
        >
          Savings
        </div>
        <div
          class="css-1mqh484-DataValue e16h0enc0"
        >
          24.00
        </div>
      </div>
      <div
        class="css-1vth247-DataCell e16h0enc2"
      >
        <div
          class="game_cell_metacritic-rating css-1e19oy8-DataItem e16h0enc1"
        >
          85
        </div>
        <div
          class="game_cell_steam-rating css-1e19oy8-DataItem e16h0enc1"
        >
          Mostly Positive
        </div>
        <div
          class="game_cell_deal-rating css-1e19oy8-DataItem e16h0enc1"
        >
          4.0
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
